File: input/paxos/sefl/acceptor.scala
val block154499 = InstructionBlock(
	mark_to_drop
)

val block154515 = InstructionBlock(
	If(Constrain('hdr.ipv4.hdrChecksum', :==:(:@('tmp_11'))), Assign('tmp_12', ConstantValue(1)), Assign('tmp_12', ConstantValue(0)))
)

val block154601 = InstructionBlock(
	Assign('hdr.ipv4.hdrChecksum', :@('tmp_13'))
)

val block147276 = InstructionBlock(
	acceptor_tbl,
	transport_tbl
)

val block156910 = InstructionBlock(
	tbl_read_round,
	tbl_act_2,
	If(Constrain(:@('tmp_22'), :==:(ConstantValue(1))), block147276)
)

val block146865 = InstructionBlock(
	mark_to_drop
)

val block146874 = InstructionBlock(
	mark_to_drop
)

val block148914 = InstructionBlock(
	Assign('tmp_15', :@('hdr.paxos.inst')),
	Assign('tmp_14', SymbolicValue()),
	Assign('meta.paxos_metadata.round', :@('tmp_14')),
	Assign('meta.paxos_metadata.set_drop', ConstantValue(1))
)

val block148992 = InstructionBlock(
	Assign('hdr.paxos.msgtype', ConstantValue(1)),
	Assign('tmp_17', :@('hdr.paxos.inst')),
	Assign('tmp_16', SymbolicValue()),
	Assign('hdr.paxos.vrnd', :@('tmp_16')),
	Assign('tmp_19', :@('hdr.paxos.inst')),
	Assign('tmp_18', SymbolicValue()),
	Assign('hdr.paxos.paxosval', :@('tmp_18')),
	Assign('tmp_20', SymbolicValue()),
	Assign('hdr.paxos.acptid', :@('tmp_20')),
	Assign('meta.paxos_metadata.set_drop', ConstantValue(0))
)

val block147033 = InstructionBlock(
	Assign('hdr.paxos.msgtype', ConstantValue(3)),
	Assign('tmp_21', SymbolicValue()),
	Assign('hdr.paxos.acptid', :@('tmp_21')),
	Assign('meta.paxos_metadata.set_drop', ConstantValue(0))
)

val block147173 = InstructionBlock(
	Assign('standard_metadata.egress_spec', :@('port')),
	Assign('hdr.udp.dstPort', :@('learnerPort'))
)

val block154858 = InstructionBlock(
	If(Constrain('hdr.paxos.rnd', :>=:(:@('meta.paxos_metadata.round'))), Assign('tmp_22', ConstantValue(1)), Assign('tmp_22', ConstantValue(0)))
)

val block51143 = InstructionBlock(

)

<Type_Error>16
<Declaration_MatchKind>106
<Declaration_MatchKind>110
<Type_Enum>167
<Type_Enum>350
val mark_to_drop = Fail('Packet dropped')

<Type_Parser>481
<Type_Control>494
<Type_Control>510
<Type_Control>526
<Type_Control>539
<Type_Control>551
<Type_Package>610
val start = InstructionBlock(
	//Extract p.ethernet
	Assign('p.ethernet.isValid', ConstantValue(1)),
	Assign('p.ethernet.dstAddr', SymbolicValue()),
	Assign('p.ethernet.srcAddr', SymbolicValue()),
	Assign('p.ethernet.etherType', SymbolicValue()),
	If(Constrain('p.ethernet.etherType', :==:(ConstantValue(2048))), parse_ipv4)
)


val parse_ipv4 = InstructionBlock(
	//Extract p.ipv4
	Assign('p.ipv4.isValid', ConstantValue(1)),
	Assign('p.ipv4.version', SymbolicValue()),
	Assign('p.ipv4.ihl', SymbolicValue()),
	Assign('p.ipv4.diffserv', SymbolicValue()),
	Assign('p.ipv4.totalLen', SymbolicValue()),
	Assign('p.ipv4.identification', SymbolicValue()),
	Assign('p.ipv4.flags', SymbolicValue()),
	Assign('p.ipv4.fragOffset', SymbolicValue()),
	Assign('p.ipv4.ttl', SymbolicValue()),
	Assign('p.ipv4.protocol', SymbolicValue()),
	Assign('p.ipv4.hdrChecksum', SymbolicValue()),
	Assign('p.ipv4.srcAddr', SymbolicValue()),
	Assign('p.ipv4.dstAddr', SymbolicValue()),
	If(Constrain('p.ipv4.protocol', :==:(ConstantValue(17))), parse_udp,
	accept)
)


val parse_udp = InstructionBlock(
	//Extract p.udp
	Assign('p.udp.isValid', ConstantValue(1)),
	Assign('p.udp.srcPort', SymbolicValue()),
	Assign('p.udp.dstPort', SymbolicValue()),
	Assign('p.udp.length_', SymbolicValue()),
	Assign('p.udp.checksum', SymbolicValue()),
	If(Constrain('p.udp.dstPort', :==:(ConstantValue(34952))), parse_paxos,
	accept)
)


val parse_paxos = InstructionBlock(
	//Extract p.paxos
	Assign('p.paxos.isValid', ConstantValue(1)),
	Assign('p.paxos.msgtype', SymbolicValue()),
	Assign('p.paxos.inst', SymbolicValue()),
	Assign('p.paxos.rnd', SymbolicValue()),
	Assign('p.paxos.vrnd', SymbolicValue()),
	Assign('p.paxos.acptid', SymbolicValue()),
	Assign('p.paxos.paxoslen', SymbolicValue()),
	Assign('p.paxos.paxosval', SymbolicValue()),
	accept
)


val accept = InstructionBlock(
	
)


val reject = InstructionBlock(
	Fail('Packet dropped')
)


val TopParser = InstructionBlock(

	//Allocate p
	Allocate('p.ethernet.isValid'),
	Assign('p.ethernet.isValid', ConstantValue(0)),
	Allocate('p.ethernet.dstAddr'),
	Allocate('p.ethernet.srcAddr'),
	Allocate('p.ethernet.etherType'),
	Allocate('p.ipv4.isValid'),
	Assign('p.ipv4.isValid', ConstantValue(0)),
	Allocate('p.ipv4.version'),
	Allocate('p.ipv4.ihl'),
	Allocate('p.ipv4.diffserv'),
	Allocate('p.ipv4.totalLen'),
	Allocate('p.ipv4.identification'),
	Allocate('p.ipv4.flags'),
	Allocate('p.ipv4.fragOffset'),
	Allocate('p.ipv4.ttl'),
	Allocate('p.ipv4.protocol'),
	Allocate('p.ipv4.hdrChecksum'),
	Allocate('p.ipv4.srcAddr'),
	Allocate('p.ipv4.dstAddr'),
	Allocate('p.udp.isValid'),
	Assign('p.udp.isValid', ConstantValue(0)),
	Allocate('p.udp.srcPort'),
	Allocate('p.udp.dstPort'),
	Allocate('p.udp.length_'),
	Allocate('p.udp.checksum'),
	Allocate('p.paxos.isValid'),
	Assign('p.paxos.isValid', ConstantValue(0)),
	Allocate('p.paxos.msgtype'),
	Allocate('p.paxos.inst'),
	Allocate('p.paxos.rnd'),
	Allocate('p.paxos.vrnd'),
	Allocate('p.paxos.acptid'),
	Allocate('p.paxos.paxoslen'),
	Allocate('p.paxos.paxosval'),

	//Allocate meta
	Allocate('meta.paxos_metadata.round'),
	Allocate('meta.paxos_metadata.set_drop'),
	Allocate('meta.paxos_metadata.ack_count'),
	Allocate('meta.paxos_metadata.ack_acceptors'),

	//Allocate standard_metadata
	Allocate('standard_metadata.ingress_port'),
	Assign('standard_metadata.ingress_port', SymbolicValue()),
	Allocate('standard_metadata.egress_spec'),
	Assign('standard_metadata.egress_spec', SymbolicValue()),
	Allocate('standard_metadata.egress_port'),
	Assign('standard_metadata.egress_port', SymbolicValue()),
	Allocate('standard_metadata.clone_spec'),
	Assign('standard_metadata.clone_spec', SymbolicValue()),
	Allocate('standard_metadata.instance_type'),
	Assign('standard_metadata.instance_type', SymbolicValue()),
	Allocate('standard_metadata.drop'),
	Assign('standard_metadata.drop', SymbolicValue()),
	Allocate('standard_metadata.recirculate_port'),
	Assign('standard_metadata.recirculate_port', SymbolicValue()),
	Allocate('standard_metadata.packet_length'),
	Assign('standard_metadata.packet_length', SymbolicValue()),

	start
)

//Control
val TopDeparser = InstructionBlock(
	//Emit hdr.ethernet
	CreateTag('hdr.ethernet.dstAddr', 0),
	Allocate(Tag('hdr.ethernet.dstAddr'), 48),
	Assign(Tag('hdr.ethernet.dstAddr'), :@('hdr.ethernet.dstAddr')),
	CreateTag('hdr.ethernet.srcAddr', 48),
	Allocate(Tag('hdr.ethernet.srcAddr'), 48),
	Assign(Tag('hdr.ethernet.srcAddr'), :@('hdr.ethernet.srcAddr')),
	CreateTag('hdr.ethernet.etherType', 96),
	Allocate(Tag('hdr.ethernet.etherType'), 16),
	Assign(Tag('hdr.ethernet.etherType'), :@('hdr.ethernet.etherType')),
	//Emit hdr.ipv4
	CreateTag('hdr.ipv4.version', 112),
	Allocate(Tag('hdr.ipv4.version'), 4),
	Assign(Tag('hdr.ipv4.version'), :@('hdr.ipv4.version')),
	CreateTag('hdr.ipv4.ihl', 116),
	Allocate(Tag('hdr.ipv4.ihl'), 4),
	Assign(Tag('hdr.ipv4.ihl'), :@('hdr.ipv4.ihl')),
	CreateTag('hdr.ipv4.diffserv', 120),
	Allocate(Tag('hdr.ipv4.diffserv'), 8),
	Assign(Tag('hdr.ipv4.diffserv'), :@('hdr.ipv4.diffserv')),
	CreateTag('hdr.ipv4.totalLen', 128),
	Allocate(Tag('hdr.ipv4.totalLen'), 16),
	Assign(Tag('hdr.ipv4.totalLen'), :@('hdr.ipv4.totalLen')),
	CreateTag('hdr.ipv4.identification', 144),
	Allocate(Tag('hdr.ipv4.identification'), 16),
	Assign(Tag('hdr.ipv4.identification'), :@('hdr.ipv4.identification')),
	CreateTag('hdr.ipv4.flags', 160),
	Allocate(Tag('hdr.ipv4.flags'), 3),
	Assign(Tag('hdr.ipv4.flags'), :@('hdr.ipv4.flags')),
	CreateTag('hdr.ipv4.fragOffset', 163),
	Allocate(Tag('hdr.ipv4.fragOffset'), 13),
	Assign(Tag('hdr.ipv4.fragOffset'), :@('hdr.ipv4.fragOffset')),
	CreateTag('hdr.ipv4.ttl', 176),
	Allocate(Tag('hdr.ipv4.ttl'), 8),
	Assign(Tag('hdr.ipv4.ttl'), :@('hdr.ipv4.ttl')),
	CreateTag('hdr.ipv4.protocol', 184),
	Allocate(Tag('hdr.ipv4.protocol'), 8),
	Assign(Tag('hdr.ipv4.protocol'), :@('hdr.ipv4.protocol')),
	CreateTag('hdr.ipv4.hdrChecksum', 192),
	Allocate(Tag('hdr.ipv4.hdrChecksum'), 16),
	Assign(Tag('hdr.ipv4.hdrChecksum'), :@('hdr.ipv4.hdrChecksum')),
	CreateTag('hdr.ipv4.srcAddr', 208),
	Allocate(Tag('hdr.ipv4.srcAddr'), 32),
	Assign(Tag('hdr.ipv4.srcAddr'), :@('hdr.ipv4.srcAddr')),
	CreateTag('hdr.ipv4.dstAddr', 240),
	Allocate(Tag('hdr.ipv4.dstAddr'), 32),
	Assign(Tag('hdr.ipv4.dstAddr'), :@('hdr.ipv4.dstAddr')),
	//Emit hdr.udp
	CreateTag('hdr.udp.srcPort', 272),
	Allocate(Tag('hdr.udp.srcPort'), 16),
	Assign(Tag('hdr.udp.srcPort'), :@('hdr.udp.srcPort')),
	CreateTag('hdr.udp.dstPort', 288),
	Allocate(Tag('hdr.udp.dstPort'), 16),
	Assign(Tag('hdr.udp.dstPort'), :@('hdr.udp.dstPort')),
	CreateTag('hdr.udp.length_', 304),
	Allocate(Tag('hdr.udp.length_'), 16),
	Assign(Tag('hdr.udp.length_'), :@('hdr.udp.length_')),
	CreateTag('hdr.udp.checksum', 320),
	Allocate(Tag('hdr.udp.checksum'), 16),
	Assign(Tag('hdr.udp.checksum'), :@('hdr.udp.checksum')),
	//Emit hdr.paxos
	CreateTag('hdr.paxos.msgtype', 336),
	Allocate(Tag('hdr.paxos.msgtype'), 16),
	Assign(Tag('hdr.paxos.msgtype'), :@('hdr.paxos.msgtype')),
	CreateTag('hdr.paxos.inst', 352),
	Allocate(Tag('hdr.paxos.inst'), 32),
	Assign(Tag('hdr.paxos.inst'), :@('hdr.paxos.inst')),
	CreateTag('hdr.paxos.rnd', 384),
	Allocate(Tag('hdr.paxos.rnd'), 16),
	Assign(Tag('hdr.paxos.rnd'), :@('hdr.paxos.rnd')),
	CreateTag('hdr.paxos.vrnd', 400),
	Allocate(Tag('hdr.paxos.vrnd'), 16),
	Assign(Tag('hdr.paxos.vrnd'), :@('hdr.paxos.vrnd')),
	CreateTag('hdr.paxos.acptid', 416),
	Allocate(Tag('hdr.paxos.acptid'), 16),
	Assign(Tag('hdr.paxos.acptid'), :@('hdr.paxos.acptid')),
	CreateTag('hdr.paxos.paxoslen', 432),
	Allocate(Tag('hdr.paxos.paxoslen'), 32),
	Assign(Tag('hdr.paxos.paxoslen'), :@('hdr.paxos.paxoslen')),
	CreateTag('hdr.paxos.paxosval', 464),
	Allocate(Tag('hdr.paxos.paxosval'), 256),
	Assign(Tag('hdr.paxos.paxosval'), :@('hdr.paxos.paxosval'))
)


//Control
val verifyChecksum = InstructionBlock(
	Allocate('tmp_11'),
	Allocate('tmp_12'),
	Assign('tmp_11', SymbolicValue()),
	tbl_act,
	If(Constrain(:@('tmp_12'), :==:(ConstantValue(1))), tbl_act_0)
)

// Action
val act = InstructionBlock(
	block154499
)


// Action
val act_0 = InstructionBlock(
	block154515
)


//Table
val tbl_act = InstructionBlock(
	Fork(act_0)
	// default_action act_0
)


//Table
val tbl_act_0 = InstructionBlock(
	Fork(act)
	// default_action act
)



//Control
val computeChecksum = InstructionBlock(
	Allocate('tmp_13'),
	Assign('tmp_13', SymbolicValue()),
	tbl_act_1
)

// Action
val act_1 = InstructionBlock(
	block154601
)


//Table
val tbl_act_1 = InstructionBlock(
	Fork(act_1)
	// default_action act_1
)



//Control
val ingress = InstructionBlock(
	Allocate('tmp_14'),
	Allocate('tmp_15'),
	Allocate('tmp_16'),
	Allocate('tmp_17'),
	Allocate('tmp_18'),
	Allocate('tmp_19'),
	Allocate('tmp_20'),
	Allocate('tmp_21'),
	Allocate('tmp_22'),
	If(Constrain('hdr.ipv4.isValid', :==:(ConstantValue(1))), If(Constrain('hdr.paxos.isValid', :==:(ConstantValue(1))), block156910))
)

// Action
val _drop_0 = InstructionBlock(
	block146865
)


// Action
val _drop_2 = InstructionBlock(
	block146874
)


// Action
val read_round_0 = InstructionBlock(
	block148914
)


// Action
val handle_1a_0 = InstructionBlock(
	block148992
)


// Action
val handle_2a_0 = InstructionBlock(
	block147033
)


//Table
val acceptor_tbl = InstructionBlock(
	// keys: hdr.paxos.msgtype
	Fork(handle_1a_0, handle_2a_0, _drop_0)
	// size ConstantValue(4)
	// default_action _drop_0
)


// Action
val forward_0 = InstructionBlock(
	Assign('port', SymbolicValue()),
	Assign('learnerPort', SymbolicValue()),
	block147173
)


//Table
val transport_tbl = InstructionBlock(
	// keys: meta.paxos_metadata.set_drop
	Fork(_drop_2, forward_0)
	// size ConstantValue(2)
	// default_action _drop_2
)


// Action
val act_2 = InstructionBlock(
	block154858
)


//Table
val tbl_read_round = InstructionBlock(
	Fork(read_round_0)
	// default_action read_round_0
)


//Table
val tbl_act_2 = InstructionBlock(
	Fork(act_2)
	// default_action act_2
)



//Control
val egress = InstructionBlock(
	place_holder_table
)

// Action
val NoAction_0 = InstructionBlock(
	block51143
)


//Table
val place_holder_table = InstructionBlock(
	Fork(NoAction_0)
	// size ConstantValue(2)
	// default_action NoAction_0
)



val main = InstructionBlock(TopParser, ingress, egress, TopDeparser)

File: experiments/case_studies/paxos/p4-16/p4src/acceptor.c
#define BITSLICE(x, a, b) ((x) >> (b)) & ((1 << ((a)-(b)+1)) - 1)
#include<stdio.h>
#include<stdint.h>
#include<stdlib.h>

int action_run;
int traverse_handle_1a = 0;
int handle_1a_msg_type = -1;
int traverse_handle_2a = 0;
int handle_2a_msg_type = -1;
int forward = 1;
uint32_t constant_acptid;



void _drop_2_163187();
void _drop_0_163171();
void transport_tbl_163485();
void acceptor_tbl_163403();
void NoAction_0_163602();
void parse_udp();
void handle_1a_0_165781();
void handle_2a_0_163324();
void accept();
void start();
void parse_paxos();
void place_holder_table_163612();
void reject();
void parse_ipv4();
void read_round_0_163194();
void forward_0_163458();
typedef struct {
	uint32_t ingress_port : 9;
	uint32_t egress_spec : 9;
	uint32_t egress_port : 9;
	uint32_t clone_spec : 32;
	uint32_t instance_type : 32;
	uint8_t drop : 1;
	uint32_t recirculate_port : 16;
	uint32_t packet_length : 32;
	uint32_t enq_timestamp : 32;
	uint32_t enq_qdepth : 19;
	uint32_t deq_timedelta : 32;
	uint32_t deq_qdepth : 19;
	uint64_t ingress_global_timestamp : 48;
	uint32_t lf_field_list : 32;
	uint32_t mcast_grp : 16;
	uint8_t resubmit_flag : 1;
	uint32_t egress_rid : 16;
} standard_metadata_t;

void mark_to_drop() {
	//printf("Packet dropped\n");
	forward = 0;
	end_assertions();
	exit(0);
}

typedef uint64_t EthernetAddress;

typedef uint32_t IPv4Address;

typedef uint32_t PortId;

typedef struct {
	uint8_t isValid : 1;
	EthernetAddress dstAddr: 48;
	EthernetAddress srcAddr: 48;
	uint32_t etherType : 16;
	uint8_t $valid$ : 1;
} ethernet_t;

typedef struct {
	uint8_t isValid : 1;
	uint8_t version : 4;
	uint8_t ihl : 4;
	uint8_t diffserv : 8;
	uint32_t totalLen : 16;
	uint32_t identification : 16;
	uint8_t flags : 3;
	uint32_t fragOffset : 13;
	uint8_t ttl : 8;
	uint8_t protocol : 8;
	uint32_t hdrChecksum : 16;
	IPv4Address srcAddr: 32;
	IPv4Address dstAddr: 32;
	uint8_t $valid$ : 1;
} ipv4_t;

typedef struct {
	uint8_t isValid : 1;
	uint32_t srcPort : 16;
	uint32_t dstPort : 16;
	uint32_t length_ : 16;
	uint32_t checksum : 16;
	uint8_t $valid$ : 1;
} udp_t;

typedef struct {
	uint8_t isValid : 1;
	uint32_t msgtype : 16;
	uint32_t inst : 32;
	uint32_t rnd : 16;
	uint32_t vrnd : 16;
	uint32_t acptid : 16;
	uint32_t paxoslen : 32;
	uint64_t paxosval : 64;
	uint8_t $valid$ : 1;
} paxos_t;

typedef struct {
	ethernet_t ethernet;
	ipv4_t ipv4;
	udp_t udp;
	paxos_t paxos;
} headers;

typedef struct {
	uint32_t round : 16;
	uint8_t set_drop : 1;
	uint8_t ack_count : 8;
	uint8_t ack_acceptors : 8;
} paxos_metadata_t;

typedef struct {
	paxos_metadata_t paxos_metadata;
} metadata;

headers hdr;
metadata meta;
standard_metadata_t standard_metadata;


void start() {
	hdr.ethernet.isValid = 1;
	switch(hdr.ethernet.etherType){
		case 2048:	parse_ipv4(); break;
		default: reject();
	}
}


void parse_ipv4() {
	hdr.ipv4.isValid = 1;
	switch(hdr.ipv4.protocol){
		case 17:	parse_udp(); break;
		default:	accept(); break;
	}
}


void parse_udp() {
	hdr.udp.isValid = 1;
	switch(hdr.udp.dstPort){
		case 34952:	parse_paxos(); break;
		default:	accept(); break;
	}
}


void parse_paxos() {
	hdr.paxos.isValid = 1;
	accept();
}


void accept() {
	
}


void reject() {
	//printf("Packet dropped");
	forward = 0;
	end_assertions();
	exit(0);
}


void TopParser() {
	klee_make_symbolic(&hdr, sizeof(hdr), "hdr");
	klee_make_symbolic(&meta, sizeof(meta), "meta");
	klee_make_symbolic(&standard_metadata, sizeof(standard_metadata), "standard_metadata");
	constant_acptid = hdr.paxos.acptid;
	start();
}

//Control

void TopDeparser() {
	//Emit hdr.ethernet
	
	//Emit hdr.ipv4
	
	//Emit hdr.udp
	
	//Emit hdr.paxos
	
}


typedef struct {
	uint8_t field : 4;
	uint8_t field_0 : 4;
	uint8_t field_1 : 8;
	uint32_t field_2 : 16;
	uint32_t field_3 : 16;
	uint8_t field_4 : 3;
	uint32_t field_5 : 13;
	uint8_t field_6 : 8;
	uint8_t field_7 : 8;
	uint32_t field_8 : 32;
	uint32_t field_9 : 32;
} tuple_0;

//Control
uint32_t tmp_6;

void verifyChecksum() {
		klee_make_symbolic(&tmp_6, sizeof(tmp_6), "tmp_6");

	if((hdr.ipv4.hdrChecksum == tmp_6)) {
	mark_to_drop();
}
}


//Control
uint32_t tmp_8;

void computeChecksum() {
		klee_make_symbolic(&tmp_8, sizeof(tmp_8), "tmp_8");

	hdr.ipv4.hdrChecksum = tmp_8;
}


//Control
uint32_t tmp_9;
uint64_t tmp_11;

void ingress() {
	if((hdr.ipv4.$valid$ == 1)) {
	if((hdr.paxos.$valid$ == 1)) {
		read_round_0_163194();
	if(hdr.paxos.rnd >= meta.paxos_metadata.round) {
		acceptor_tbl_163403();
	transport_tbl_163485();

}

}
}
}

// Action
void _drop_0_163171() {
	action_run = 163171;
		mark_to_drop();

}


// Action
void _drop_2_163187() {
	action_run = 163187;
		mark_to_drop();

}


// Action
void read_round_0_163194() {
	action_run = 163194;
	uint32_t tmp_round;
	klee_make_symbolic(&tmp_round, sizeof(tmp_round), "tmp_round");
	meta.paxos_metadata.round = tmp_round;

	meta.paxos_metadata.set_drop = 1;

}


// Action
void handle_1a_0_165781() {
	traverse_handle_1a = 1;
	handle_1a_msg_type = (hdr.paxos.msgtype == 0);
	action_run = 165781;
		hdr.paxos.msgtype = 1;
		klee_make_symbolic(&tmp_9, sizeof(tmp_9), "tmp_9");

	hdr.paxos.vrnd = tmp_9;
		klee_make_symbolic(&tmp_11, sizeof(tmp_11), "tmp_11");

	hdr.paxos.paxosval = tmp_11;
	uint32_t tmp_acptid;
	klee_make_symbolic(&tmp_acptid, sizeof(tmp_acptid), "tmp_acptid");
	hdr.paxos.acptid = tmp_acptid;

	meta.paxos_metadata.set_drop = 0;

}


// Action
void handle_2a_0_163324() {
	traverse_handle_2a = 1;
	handle_2a_msg_type = (hdr.paxos.msgtype == 2);
	action_run = 163324;
		hdr.paxos.msgtype = 3;
	uint32_t tmp_acptid;
	klee_make_symbolic(&tmp_acptid, sizeof(tmp_acptid), "tmp_acptid");
	hdr.paxos.acptid = tmp_acptid;

	meta.paxos_metadata.set_drop = 0;

}


// Action
void forward_0_163458() {
	action_run = 163458;
	PortId port;
	klee_make_symbolic(&port, sizeof(port), "port");
uint32_t learnerPort;
	klee_make_symbolic(&learnerPort, sizeof(learnerPort), "learnerPort");
	standard_metadata.egress_spec = port;
	hdr.udp.dstPort = learnerPort;

}


//Table
void acceptor_tbl_163403() {
	// keys: hdr.paxos.msgtype:exact
	int symbol;
	klee_make_symbolic(&symbol, sizeof(symbol), "symbol");
	switch(symbol) {
		case 0: handle_1a_0_165781(); break;
		case 1: handle_2a_0_163324(); break;
		default: _drop_0_163171(); break;
	}
	// size 4
	// default_action _drop_0();

}


//Table
void transport_tbl_163485() {
	// keys: meta.paxos_metadata.set_drop:exact
	int symbol;
	klee_make_symbolic(&symbol, sizeof(symbol), "symbol");
	switch(symbol) {
		case 0: _drop_2_163187(); break;
		default: forward_0_163458(); break;
	}
	// size 2
	// default_action _drop_2();

}



//Control

void egress() {
	place_holder_table_163612();
}

// Action
void NoAction_0_163602() {
	action_run = 163602;
	
}


//Table
void place_holder_table_163612() {
	int symbol;
	klee_make_symbolic(&symbol, sizeof(symbol), "symbol");
	switch(symbol) {
		default: NoAction_0_163602(); break;
	}
	// size 2
	// default_action NoAction_0();

}

void end_assertions(){

	if(traverse_handle_1a && !handle_1a_msg_type && (handle_1a_msg_type != -1)){
		klee_warning_once("handle_1a executed with msgtype different that 1A");
		klee_assert(0);
	}

	if(traverse_handle_2a && !handle_2a_msg_type && (handle_2a_msg_type != -1)){
		klee_warning_once("handle_2a executed with msgtype different that 2A");
		klee_assert(0);
	}

	if(forward && (constant_acptid == hdr.paxos.acptid)){
		klee_warning_once("packet forwarded and swid not updated");
		klee_assert(0);
	}

	if(forward && !(traverse_handle_1a || traverse_handle_2a)){
		klee_warning("WTH???");
		klee_assert(0);
	}
}

int main() {
	TopParser();
	ingress();
	egress();
	TopDeparser();
	end_assertions();
	return 0;
}


